include/driver/target.h

```diff
@@ -40,26 +45,41 @@ class CPU : public Target {
 };
 
 class GPU : public Target {
-    Opt<std::pair<int, int>> computeCapability_;
+#ifdef FT_WITH_CUDA
+    Ref<cudaDeviceProp> infoArch_;
+#endif // FT_WITH_CUDA
 
   public:
+#ifndef FT_WITH_CUDA
     GPU(bool useNativeArch = true) : Target(useNativeArch) {}
```

Comment:
By design, a `Target` can contain information of a local device only when the flag `useNativeArch` is `true`. However, after this PR, all GPU `Target`s can only be constructed from local devices.

I suggest moving `useNativeArch` to local to `CPU`. When constructed from a CPU `Device`, set it to true. Otherwise, let users configure it.